Api/WebhookController.cs

```diff
@@ -0,0 +1,146 @@
+using System.Globalization;
+using System.Net.Http.Headers;
+using System.Xml;
+using Microsoft.AspNetCore.Mvc;
+using Newtonsoft.Json;
+using Twilio.AspNet.Core;
+using Twilio.TwiML;
+
+namespace TwilioSmsScheduler.Api;
+
+[ApiController]
+[Route("api/[controller]")]
+public class WebhookController : ControllerBase
+{
+    private readonly ILogger<WebhookController> logger;
+    private readonly HttpClient httpClient;
+    private readonly string clientId, clientSecret, requestTokenBaseUrl, calendarUrl;
+
+    public WebhookController(IConfiguration configuration, ILogger<WebhookController> logger, HttpClient httpClient)
+    {
+        this.logger = logger;
+        this.httpClient = httpClient;
+        clientId = configuration["GoogleApi:ClientID"];
+        clientSecret = configuration["GoogleApi:Secret"];
+        requestTokenBaseUrl = configuration["GoogleApi:RequestTokenUrl"];
+        calendarUrl = configuration["GoogleApi:CalendarUrl"];
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> IncomingMessage()
+    {
+        string messageBody = Request.Form["Body"];
+        string messageFrom = Request.Form["From"];
+
+        var userCalendarConfig = UserCalendarConfiguration.Instance;
+
+        // Extract appointment
+        var utcOffset = TimeZoneInfo.Local.BaseUtcOffset;
+        var timeZone = "UTC" + (utcOffset > TimeSpan.Zero ? "+" : "-") + utcOffset.ToString(@"h\:mm");
```

Comment:
Use the current machine's timezone instead of hard-coding. Before this code change, I could keep scheduling appointments at the same times on the same days even though they should be conflicting.
Because of the timezone mismatch of the timezone used in Google Calendar vs this timezone, it wouldn't be conflicting in the code when it should.